### ISSUE TITLE
ComboBox source is IList but can only be list of strings

### ISSUE THREAD

**Issue by nanocortex** [migueldeicaza/gui.cs#746]
If `ComboBox` source is set as list of objects, e.g.:

```csharp

public class CategoryItem
{
	public CategoryItem(int id, string name)
	{
		Id = id;
		Name = name;
	}
	public int Id { get; set; }
	public string Name { get; set; }

	public override string ToString()
	{
		return Name;
	}
}

var categoryComboBox = new ListView() {
	X = 1,
	Y = 1,
	Width = Dim.Fill (),
	Height = Dim.Fill (2),
};

categoryComboBox.SetSource (new List<CategoryItem> () {
	new CategoryItem (1, "Category"),
	new CategoryItem (2, "Other category")
});

Win.Add (categoryComboBox);

```

it's throwing exception when searching, becuse of a string cast on this line:
https://github.com/migueldeicaza/gui.cs/blob/527c8aa36526e0e0347cecae7ecd66196ee5aae4/Terminal.Gui/Views/ComboBox.cs#L327

I understand that it's due to a fact that `ComboBox` item can be any `string` value, not like in `ListView` where you can select values only from `DataSource`.

My use case is that I want to choose from predefined set of values with search capability. Is it something `ComboBox` should do or it's rather an idea for a new control like `DropDownList`?

**Comment by BDisp** [comment]
There is a pending PR #710. Merge and check if the problem persists.

**Comment by nanocortex** [comment]
Thanks, change in PR fixed it for me. Although my other conern is how to know which object is selected?
In case of `ListView` I can just use `SelectedItem` as index of existing list of objects, but ComboBox doesn't have such property and `ListViewItemEventArgs.Item` is not an index of whole collection but rather index of currently narrowed down list.

What would be the best approach for this?

**Comment by BDisp** [comment]
I really asked that question and the control should really have that property with the index based on the collection of the whole list and I even presented an option for that purpose but the author of the control is still analyzing it.

**Comment by fergusonr** [comment]
Hi, SelectedItem has now been implemented in PR #710. Its read-only. It does not make any sense to set it unless the full source  list is displayed. Something to consider in the future.

**Comment by BDisp** [comment]
Hi. Thanks. SelectedItem is required to appear selected in the list if an item is already selected in the database.

**Comment by fergusonr** [comment]
Please assign to me

**Comment by BDisp** [comment]
@fergusonr I think this is already fixed.

**Comment by fergusonr** [comment]
> @fergusonr I think this is already fixed.

This issues 'Title' is fixed but not your last comment "SelectedItem is required to appear selected"

**Comment by BDisp** [comment]
I already fixed it, see my PR #751.

**Comment by tig** [comment]
Sweet.
